packages/perseus-editor/src/components/graph-locked-figures/locked-function-settings.tsx

```diff
@@ -0,0 +1,246 @@
+import LineSwatch from "./line-swatch";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {LockedFigureSettingsCommonProps} from "./locked-figure-settings";
+import type {LockedFunctionType} from "@khanacademy/perseus";
+import type {Interval} from "mafs";
+
+export type Props = LockedFunctionType &
+    LockedFigureSettingsCommonProps & {
+        /**
+         * Called when the props (points, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedFunctionType>) => void;
+    };
+
+const LockedFunctionSettings = (props: Props) => {
+    const {
+        color: lineColor,
+        strokeStyle,
+        equation,
+        directionalAxis,
+        domain,
+        onChangeProps,
+        onMove,
+        onRemove,
+    } = props;
+    const equationPrefix = directionalAxis === "x" ? "y=" : "x=";
+    const lineLabel = `Function (${equationPrefix}${equation})`;
+    const domainLimits = domain ? [...domain] : [-Infinity, Infinity];
+
+    // Tracking the string value of domain values to handle interim state of
+    //     entering in a negative value - used when specifying value of input field
+    const [domainEntries, setDomainEntries] = useState([
+        domain && domain[0] !== -Infinity ? domain[0].toString() : "",
+        domain && domain[1] !== Infinity ? domain[1].toString() : "",
+    ]);
+
+    useEffect(() => {
+        setDomainEntries([
+            domain && domain[0] !== -Infinity ? domain[0].toString() : "",
+            domain && domain[1] !== Infinity ? domain[1].toString() : "",
+        ]);
+    }, [domain]);
+
+    // Generic function for handling property changes (except for 'domain')
+    function handlePropChange(property: string, newValue: string) {
+        const updatedProps: Partial<LockedFunctionType> = {};
+        updatedProps[property] = newValue;
+        onChangeProps(updatedProps);
+    }
+
+    /*
+     Reason for having a separate 'propChange' function for 'domain':
+        Domain entries are optional. Their default value is +/- Infinity.
+        Since input fields that are empty evaluate to zero, there needs to be
+            dedicated code to convert empty to Infinity.
+     */
+    function handleDomainChange(limitIndex: number, newValueString: string) {
+        const newDomainEntries = [...domainEntries];
+        newDomainEntries[limitIndex] = newValueString;
+        setDomainEntries(newDomainEntries);
+        const newDomain: Interval | undefined = [
+            domainLimits[0],
+            domainLimits[1],
+        ];
+
+        let newValue = parseFloat(newValueString);
+        if (newValueString === "" && limitIndex === 0) {
+            newValue = -Infinity;
+        } else if (newValueString === "" && limitIndex === 1) {
+            newValue = Infinity;
+        }
+        newDomain[limitIndex] = newValue;
+        onChangeProps({domain: newDomain});
+    }
+
+    return (
+        <UniqueIDProvider
```

Comment:
Now that we're on React 18, I think you can use [React.useId()](https://react.dev/reference/react/useId) instead of the unique ID provider.

Reply:
Good to know - thanks!

In reviewing this line, I realized that the need for a unique ID is no longer in place, and I just forgot to remove it.